force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+        Contact con1 = UTIL_UnitTestData_TEST.getContact();
+        Contact con2 = UTIL_UnitTestData_TEST.getContact();
+        con1.MailingStreet = 'Some Street';
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(con1, con2);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChanged() {
+        /* return (!isContactAddressEmpty(con1) || String.isNotBlank(con1.Primary_Address_Type__c)); does not make any sense
+         or hardly readable. We need to refactor this so that it is readable from first glance */
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isAccountAddressChanged *******************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void testIsAccountAddressChangedAcc2NullFalse() {
+        Account acc1 = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(acc1, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedAcc1NullFalse() {
+        Account acc2 = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(null, acc2);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedNotNullSameAddressInfoFalse() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            2,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(accounts[0], accounts[1]);
+        Test.stopTest();
+
+        System.assertEquals(false, isAccountAddressChanged);
+    }
+
+    @isTest
+    static void testIsAccountAddressChangedNotNullDifferentAddressInfoTrue() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            2,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+        accounts[0].BillingStreet = 'some other billing street';
+
+        Test.startTest();
+        Boolean isAccountAddressChanged = ADDR_Addresses_UTIL.isAccountAddressChanged(accounts[0], accounts[1]);
+        Test.stopTest();
+
+        System.assertEquals(true, isAccountAddressChanged);
+    }
+
```

Comment:
We don't have coverage on the state and country code picklists, and we can only test for when they're turned off, based on org config when we get the ability to override. So make a note in the class about the testing gap.

Reply:
I left the comments in the unit test class.